src/renderer/components/dock/dock.tsx

```diff
@@ -75,6 +87,19 @@ class NonInjectedDock extends React.Component<DockProps & Dependencies> {
     this.element?.current.focus();
   };
 
+  switchToNextTab = (direction: Direction = Direction.NEXT) => {
+    const { tabs, selectedTab } = this.props.dockStore;
+    const currentIndex = tabs.indexOf(selectedTab);
+    const nextIndex = currentIndex + direction;
+
+    // check if moving to the next or previous tab is possible.
+    if (currentIndex!== -1 && (nextIndex >= tabs.length || nextIndex < 0)) return;
```

Comment:
Should also return if `currentIndex === -1`?

```suggestion
    if (currentIndex === -1 || (nextIndex >= tabs.length || nextIndex < 0)) return;
```